dask_jobqueue/core.py

```diff
@@ -520,10 +525,23 @@ def __init__(
             "options": scheduler_options,
         }
 
+        if shared_directory is None:
+            shared_directory = dask.config.get(
+                "jobqueue.%s.shared-directory" % config_name
+            )
+        self.shared_directory = shared_directory
+
+        if security is not None:
+            if shared_directory is None:
+                raise ValueError(
+                    "Passing a security object to workers is only supported if a shared directory is configured"
+                )
```

Comment:
```suggestion
        if security is not None and shared_directory is None:
            raise ValueError(
                "Passing a security object to workers is only supported if a shared directory is configured"
            )
```

Reply:
Feel free to ignore this, but since this is a new feature it might be helpful to make this error message a little verbose to include pointers on how to set this shared directory. e.g. 

```python
raise ValueErrror("Passing a security object to workers is only supported if a shared directory is configured. \
                  You can set this value by using dask for e.g. `dask.config.set({'jobqueue.pbs.shared_directory': '/home/foo'})` or by setting this value in the config file found in `~/.config/dask/jobqueue.yaml` ")
```

Reply:
tI like this. Actually it made me think that it should be an warning at this point because there is the rare case that certificates can be predeployed on the compute nodes. So I will a) make this verbose and b) make it a warning and only later an error if temp certificates are used. The basic problem is that the actual error a user will encounter is very much hidden and only will lead to timeouts because no workers are started.

Reply:
Please consider @andersy005 suggestion here!

Also, we might decide to specify a default even if this is None (something like: use the current working directory), so maybe a warning would be better.

Reply:
Yes I will do a warning and make this more narrow. Actually this covers cases also where there is a fixed cert given.